myriad-executor/src/main/java/com/ebay/myriad/executor/MyriadExecutor.java

```diff
@@ -124,152 +60,28 @@ public void disconnected(ExecutorDriver driver) {
-            envNMOptions += String.format(PROPERTY_FORMAT, KEY_YARN_NM_CONTAINER_EXECUTOR_CLASS, VAL_YARN_NM_CONTAINER_EXECUTOR_CLASS);
-            envNMOptions += String.format(PROPERTY_FORMAT, KEY_YARN_NM_LCE_RH_CLASS, VAL_YARN_NM_LCE_RH_CLASS);
-
-            String containerId = getContainerId();
-
-            makeWritable("/sys/fs/cgroup/cpu/mesos/" + containerId);
-
-            // TODO: Configure hierarchy
-            envNMOptions += String.format(PROPERTY_FORMAT, KEY_YARN_NM_LCE_CGROUPS_HIERARCHY, "mesos/" + containerId);
-            envNMOptions += String.format(PROPERTY_FORMAT, KEY_YARN_NM_LCE_CGROUPS_MOUNT, "true");
-            // TODO: Make it configurable
-            envNMOptions += String.format(PROPERTY_FORMAT, KEY_YARN_NM_LCE_CGROUPS_MOUNT_PATH, "/sys/fs/cgroup");
-            envNMOptions += String.format(PROPERTY_FORMAT, KEY_YARN_NM_LCE_GROUP, "root");
-            envNMOptions += String.format(PROPERTY_FORMAT, KEY_YARN_HOME, taskConfig.getYarnEnvironment().get("YARN_HOME"));
-        } else {
-            // Otherwise configure to use Default
-            envNMOptions += String.format(PROPERTY_FORMAT, KEY_YARN_NM_CONTAINER_EXECUTOR_CLASS, DEFAULT_YARN_NM_CONTAINER_EXECUTOR_CLASS);
-        }
-        envNMOptions += String.format(PROPERTY_FORMAT, KEY_NM_RESOURCE_CPU_VCORES, taskConfig.getAdvertisableCpus() + "");
-        envNMOptions += String.format(PROPERTY_FORMAT, KEY_NM_RESOURCE_MEM_MB, taskConfig.getAdvertisableMem() + "");
-        envNMOptions += String.format(PROPERTY_FORMAT, KEY_NM_ADDRESS, "0.0.0.0:" + taskConfig.getRpcPort().toString() + "");
-        envNMOptions += String.format(PROPERTY_FORMAT, KEY_NM_LOCALIZER_ADDRESS, "0.0.0.0:" + taskConfig.getLocalizerPort().toString() + "");
-        envNMOptions += String.format(PROPERTY_FORMAT, KEY_NM_WEBAPP_ADDRESS, "0.0.0.0:" + taskConfig.gettWebAppHttpPort().toString() + "");
-        envNMOptions += String.format(PROPERTY_FORMAT, KEY_NM_SHUFFLE_PORT, taskConfig.getShufflePort().toString() + "");
-
-        return envNMOptions;
-    }
-
-    public String getContainerId() {
-        String cwd = Paths.get(".").toAbsolutePath().normalize().toString();
-        String[] split = cwd.split("/");
-        return split[split.length - 1];
+      TaskStatus status = TaskStatus.newBuilder()
+        .setTaskId(task.getTaskId())
+        .setState(TaskState.TASK_RUNNING)
+        .build();
+      driver.sendStatusUpdate(status);
     }
 
     @Override
     public void killTask(ExecutorDriver driver, TaskID taskId) {
         LOGGER.debug("KillTask received for taskId: " + taskId.getValue());
-        this.process.destroy();
+
         TaskStatus status = TaskStatus.newBuilder()
                 .setTaskId(taskId)
                 .setState(TaskState.TASK_KILLED)
                 .build();
         driver.sendStatusUpdate(status);
+        throw new RuntimeException("NodeManager shutdown after receiving" +
+          " KillTask for taskId " + taskId.getValue());
     }
```

Comment:
yes, you are absolutely correct. Missed this one. Will make sure that resources are returned back to Mesos for all the placeholder tasks and then the NM task is marked TASK_FINISHED.